src/olympia/discovery/tests/test_admin.py

```diff
@@ -546,6 +551,164 @@ def test_query_count(self):
         assert response.status_code == 200
 
 
+class TestPrimaryHeroImageAdmin(TestCase):
+    def setUp(self):
+        self.list_url = reverse(
+            'admin:discovery_primaryheroimageupload_changelist')
+        self.detail_url_name = 'admin:discovery_primaryheroimageupload_change'
+
+    def _get_heroform(self, item_id):
+        return {
+            "custom_image": "",
+        }
+
+    def test_can_see_primary_hero_image_in_admin_with_discovery_edit(self):
+        user = user_factory()
+        self.grant_permission(user, 'Admin:Tools')
+        self.grant_permission(user, 'Discovery:Edit')
+        self.client.login(email=user.email)
+        url = reverse('admin:index')
+        response = self.client.get(url)
+        assert response.status_code == 200
+
+        # Use django's reverse, since that's what the admin will use. Using our
+        # own would fail the assertion because of the locale that gets added.
+        self.list_url = django_reverse(
+            'admin:discovery_primaryheroimageupload_changelist')
+        assert self.list_url in response.content.decode('utf-8')
+
+    def test_can_list_with_discovery_edit_permission(self):
+        PrimaryHeroImage.objects.create(custom_image='föo.jpg')
+        user = user_factory()
+        self.grant_permission(user, 'Admin:Tools')
+        self.grant_permission(user, 'Discovery:Edit')
+        self.client.login(email=user.email)
+        response = self.client.get(self.list_url, follow=True)
+        assert response.status_code == 200
+        assert 'föo.jpg' in response.content.decode('utf-8')
+
+    def test_can_edit_with_discovery_edit_permission(self):
+        item = PrimaryHeroImage.objects.create(custom_image='föo.jpg')
+        self.detail_url = reverse(
+            'admin:discovery_primaryheroimageupload_change', args=(item.pk,)
+        )
+        user = user_factory()
+        self.grant_permission(user, 'Admin:Tools')
+        self.grant_permission(user, 'Discovery:Edit')
+        self.client.login(email=user.email)
+        response = self.client.get(self.detail_url, follow=True)
+        assert response.status_code == 200
+        content = response.content.decode('utf-8')
+        assert u'föo.jpg' in content
+
+        response = self.client.post(
+            self.detail_url,
+            dict(self._get_heroform(str(item.id)), **{
+                'custom_image': 'föo.jpg',
```

Comment:
this should be changing the `custom_image` value to verify the edit is working

Reply:
Oy, apologies for the oversight! I got caught up in making sure I included all the tests and checking they all passed *facepalm. I have updated the value and have it uploading a new file data instead of setting the filename.